mlir/include/mlir/Dialect/Bufferization/IR/BufferizationOps.td

```diff
@@ -496,7 +501,8 @@ def Bufferization_ToMemrefOp : Bufferization_Op<"to_memref", [
     Pure,
     TypesMatchWith<"type of 'tensor' is the tensor equivalent of 'memref'",
                    "memref", "tensor",
-                   "memref::getTensorTypeFromMemRefType($_self)">
+                   "memref::getTensorTypeFromMemRefType($_self)",
+                   "bufferization::detail::tensorTypesMatchUpToEncoding">
```

Comment:
I think you can use `AllShapesMatch` and `AllElementTypesMatch` instead of `tensorTypesMatchUpToEncoding`.

Reply:
Fixed